tests/test_functional.py

```diff
@@ -48,6 +47,11 @@ class TestDialect(csv.excel):
 
     csv.register_dialect("test", TestDialect)
 
+    def _get_expected(self):
+        with self._open_source_file() as f:
+            expected_msgs = self.get_expected_messages(f)
+        return expected_msgs, []
+
```

Comment:
I do not find the call place of this method...

Reply:
l.72 and l.106 in `tests/test_func.py` and l. 167 in `pylint/testutils/lint_module_test.py` :)

Reply:
Actually the instance is initialized here:
```python
@pytest.mark.parametrize("test_file", TESTS, ids=TESTS_NAMES)
def test_functional(test_file, recwarn):
    if UPDATE_FILE.exists():
        lint_test = LintModuleOutputUpdate(test_file)
    else:
        lint_test = testutils.LintModuleTest(test_file)
```
If we're using the update we instantiate a `LintModuleOutputUpdate` that will be using this function later.

Reply:
@Pierre-Sassoulas i'm sorry but i don't get it.
It seems to me that in `tests/test_func.py` l.72 and l.106 it calls the method defined in the same module at line 93.
The call at l167 in `lint_module_test.py` seems to be related to the method defined at line 140 of the same module.
I'm i wrong?

Reply:
Regarding the call line  167 in `lint_module_test.py` It the update file exists we instantiate a child class `LintModuleOutputUpdate` of the mother class `LintModuleTest` (code in comment above, from test_functional.py).  Then when reaching `_get_expected` l167 what is called is not the one defined l140 in LintModuleTest the but the one we're talking about overidden in the child class. Basically it comes done to this snippet with Mother=LintModuleTest and Child=LintModuleOutputUpdate:
```python
class Mother:
    
    def __init__(self):
        print("Get expected is : {}".format(self._get_expected()))
        
    def _get_expected(self):
        return "Mother implementation."

class Child(Mother):
    
        def _get_expected(self):
            return "Child implementation."


Mother() # Get expected is : Mother implementation.                                                                                                                               
Child() # Get expected is : Child implementation.   
```

Reply:
Ok! Thanks @Pierre-Sassoulas